### PR TITLE
feat: Add ability to define custom timeout for fargate profiles

### PR DESCRIPTION
**Pull request by IvanDechovsky**
Add ability to define custom timeout for create/delete operations for fargate profiles

https://github.com/terraform-aws-modules/terraform-aws-eks/issues/1610

### CODE REVIEW

File: examples/complete/main.tf
Comment (daroga0002) [inline examples/complete/main.tf:100]:
```
Error: Error parsing "create" timeout: time: missing unit in duration "15"
```

File: examples/fargate/main.tf
Comment (daroga0002) [inline examples/fargate/main.tf:56]:
```
Error: Error parsing "create" timeout: time: missing unit in duration "15"
```

File: examples/fargate/main.tf
Comment (daroga0002) [inline examples/fargate/main.tf:121]:
```
Error: Error parsing "create" timeout: time: missing unit in duration "15"
```

File: modules/fargate/README.md
Comment (daroga0002) [inline modules/fargate/README.md:14]: does those changes where done by you or by some automation?

File: modules/fargate/README.md
Comment (daroga0002) [inline modules/fargate/README.md:66]: what has done those changes?

this part of file should be autogenerated via pre commit hooks

File: modules/fargate/main.tf
Comment (daroga0002) [inline modules/fargate/main.tf:67]: instead assuming some values put failover to `null` value which will use provider defaults

Comment (daroga0002) [reply modules/fargate/main.tf:67]:
```suggestion
    create = lookup(each.value["timeouts"], "create", null)
```

File: modules/fargate/main.tf
Comment (daroga0002) [inline modules/fargate/main.tf:68]:
```suggestion
    delete = lookup(each.value["timeouts"], "delete", null)
```

**Review by daroga0002 — CHANGES_REQUESTED** [review]
This change seems be not tested, please test and fix a comments


Additionally I don`t know does changes in README.md where done by you or any automatic ?? (we use precommit hooks to generate some part of docs using terraform-docs)

File: variables.tf
Comment (daroga0002) [inline variables.tf:390]: We dont want add additionall variables into submodule, it should follow previously used pattern. I dont know why you changed this as I given you direct suggestion how to make default null value

**Review by daroga0002 — CHANGES_REQUESTED** [review]
We should same pattern across timeouts (was used previously in this PR, I dont know why you changed)

**Comment by IvanDechovsky** [comment]
@daroga0002 Will you have some time to review it this week perhaps?

**Comment by IvanDechovsky** [comment]
Any chance this can be looked at please?

**Comment by daroga0002** [comment]
I have this on my list, will do this till end of week

File: examples/fargate/main.tf
Comment (daroga0002) [inline examples/fargate/main.tf:172]: lets remove `create` just for test case when we have just one subargument

File: examples/fargate/main.tf
Comment (daroga0002) [inline examples/fargate/main.tf:99]: remove lines 99-103 for test case when we dont define any timeout arguments

**Review by daroga0002 — CHANGES_REQUESTED** [review]
Please add into file 
https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/modules/fargate/README.md

in line 17 following content:
```
| timeouts | A map of timeouts for create/delete operations. | `map(string)` | Provider default behavior | no |
```

**Comment by daroga0002** [comment]
@IvanDechovsky thank you for your contribution, 3 comments and after fixing we will be able to merge it

**Comment by IvanDechovsky** [comment]
@daroga0002 Thank you for taking a look. Updated!

File: modules/fargate/README.md
Comment (daroga0002) [inline modules/fargate/README.md:18]:
```suggestion
| timeouts | A map of timeouts for create/delete operations. | `map(string)` | Provider default behaviour | no |
```

**Review by daroga0002 — APPROVED** [review]
One typo in `behaviour` to fix and we can merge.

@antonbabenko  this can be merged when typo will be fixed (maybe you can accept commit on suggestion?)

File: modules/fargate/README.md
Comment (antonbabenko) [reply modules/fargate/README.md:18]: US vs UK :) Let's leave the English language to native-language speakers (definitely, not me and you)

**Comment by antonbabenko** [comment]
This PR is included in version 17.24.0 :tada:

**Pull request merged**
